nuxeo-core/nuxeo-core-mongodb/src/main/java/org/nuxeo/ecm/core/mongodb/kv/MongoDBKeyValueStore.java

```diff
@@ -181,6 +186,67 @@ protected Object getObject(String key) {
         return value;
     }
 
+    /**
+     * @since 9.10
+     */
+    @Override
+    public Map<String, byte[]> get(Set<String> keys) {
+        Objects.requireNonNull(keys);
+        Map<String, byte[]> kvMap = new HashMap<>(keys.size());
+        Block<Document> block = doc -> {
+            String key = doc.getString(ID_KEY);
+            Object value = doc.get(VALUE_KEY);
+            byte[] bytesValue;
+            if (value instanceof String) {
+                bytesValue = ((String) value).getBytes(UTF_8);
+            } else if (value instanceof Binary) {
+                bytesValue = ((Binary) value).getData();
+            } else {
```

Comment:
What about missing values (i.e., `value == null`). Please add a unit test for that and treat that case.